File: Projects/04/Clojure/src/factorial.clj
;;;; factorial.clj computes factorials
;;;;
;;;; Input: a number
;;;; Precondition: the number must be positive
;;;; Output: the factorial of the number
;;;;
;;;; Completed by: AJ Vrieland (ajv234)
;;;; Date: 03/01/2020
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns factorial)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; -main computes the factorial with a given number
;;; Input: a number
;;; Output: the factorial of that number
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn -main []
  (print "Enter Factorial: ") (flush)
  (let
    [ number (double (read)) ]
    (def answer 1)
    (loop [i 2]
      (when (<= i number)
        (def answer (* answer i))
        (recur (+ i 1))
      )
    )
    (println answer)
  )
)

File: Projects/04/Ruby/factorial.rb
# factorial.rb computes the factorial of a number
#
# Completed by: AJ Vrieland (ajv234)
# Date: 03/01/2020
#
# Input: Factorial number
# Precondition: the number must be positive
# Output: The factorial of that number


if __FILE__ == $0
  print "Enter factorial: "
  num = gets.chop.to_f
  answer = 1.0
  increment = 2.0

  while (increment <= num)
    answer *= increment
    increment += 1
  end

  print answer
  print "\n"

end
